### PR TITLE
Cspell: SPELL_CSPELL_ANALYZE_FILE_NAMES

### PR DESCRIPTION
**Pull request by nvuillam**
Allow cspell to also analyze file names
Fixes https://github.com/oxsecurity/megalinter/issues/2009

### CODE REVIEW

**Comment by nvuillam** [comment]
## [🦙 MegaLinter](https://oxsecurity.github.io/megalinter/beta) status: :warning: [**WARNING**](https://github.com/oxsecurity/megalinter/actions/runs/3357733083)

|    Descriptor    |                                                        Linter                                                        |Files|Fixed|                                  Errors                                   |Elapsed time|
|------------------|----------------------------------------------------------------------------------------------------------------------|-----|----:|---------------------------------------------------------------------------|------------|
|✅ BASH           |[bash-exec](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_bash_exec)                                  |    7|     |                                                                          0|0.01s       |
|✅ BASH           |[shellcheck](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_shellcheck)                                |    7|     |                                                                          0|0.27s       |
|✅ BASH           |[shfmt](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_shfmt)                                          |    7|    0|                                                                          0|0.3s        |
|✅ COPYPASTE      |[jscpd](https://oxsecurity.github.io/megalinter/beta/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                         |1.85s       |
|✅ DOCKERFILE     |[hadolint](https://oxsecurity.github.io/megalinter/beta/descriptors/dockerfile_hadolint)                              |  105|     |                                                                          0|8.03s       |
|✅ JSON           |[eslint-plugin-jsonc](https://oxsecurity.github.io/megalinter/beta/descriptors/json_eslint_plugin_jsonc)              |   21|    0|                                                                          0|2.07s       |
|✅ JSON           |[jsonlint](https://oxsecurity.github.io/megalinter/beta/descriptors/json_jsonlint)                                    |   19|     |                                                                          0|0.23s       |
|✅ JSON           |[v8r](https://oxsecurity.github.io/megalinter/beta/descriptors/json_v8r)                                              |   21|     |                                                                          0|17.41s      |
|:warning: MARKDOWN|[markdownlint](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdownlint)                        |  294|    0|[**5**](https://github.com/oxsecurity/megalinter/actions/runs/3357733083)  |5.2s        |
|✅ MARKDOWN       |[markdown-link-check](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdown_link_check)          |  294|     |                                                                          0|5.25s       |
|✅ MARKDOWN       |[markdown-table-formatter](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdown_table_formatter)|  294|    0|                                                                          0|7.47s       |
|✅ OPENAPI        |[spectral](https://oxsecurity.github.io/megalinter/beta/descriptors/openapi_spectral)                                 |    1|     |                                                                          0|0.8s        |
|:warning: PYTHON  |[bandit](https://oxsecurity.github.io/megalinter/beta/descriptors/python_bandit)                                      |  170|     |[**42**](https://github.com/oxsecurity/megalinter/actions/runs/3357733083) |2.4s        |
|✅ PYTHON         |[black](https://oxsecurity.github.io/megalinter/beta/descriptors/python_black)                                        |  170|    0|                                                                          0|4.01s       |
|✅ PYTHON         |[flake8](https://oxsecurity.github.io/megalinter/beta/descriptors/python_flake8)                                      |  170|     |                                                                          0|2.06s       |
|✅ PYTHON         |[isort](https://oxsecurity.github.io/megalinter/beta/descriptors/python_isort)                                        |  170|    0|                                                                          0|0.67s       |
|✅ PYTHON         |[mypy](https://oxsecurity.github.io/megalinter/beta/descriptors/python_mypy)                                          |  170|     |                                                                          0|7.36s       |
|✅ PYTHON         |[pylint](https://oxsecurity.github.io/megalinter/beta/descriptors/python_pylint)                                      |  170|     |                                                                          0|11.29s      |
|:warning: PYTHON  |[pyright](https://oxsecurity.github.io/megalinter/beta/descriptors/python_pyright)                                    |  170|     |[**274**](https://github.com/oxsecurity/megalinter/actions/runs/3357733083)|15.7s       |
|✅ REPOSITORY     |[checkov](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_checkov)                                |yes  |     |no                                                                         |27.85s      |
|✅ REPOSITORY     |[git_diff](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_git_diff)                              |yes  |     |no                                                                         |0.3s        |
|✅ REPOSITORY     |[secretlint](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_secretlint)                          |yes  |     |no                                                                         |8.64s       |
|✅ REPOSITORY     |[trivy](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_trivy)                                    |yes  |     |no                                                                         |20.89s      |
|✅ SPELL          |[cspell](https://oxsecurity.github.io/megalinter/beta/descriptors/spell_cspell)                                       |  707|     |                                                                          0|17.66s      |
|✅ SPELL          |[misspell](https://oxsecurity.github.io/megalinter/beta/descriptors/spell_misspell)                                   |  533|    0|                                                                          0|0.73s       |
|✅ XML            |[xmllint](https://oxsecurity.github.io/megalinter/beta/descriptors/xml_xmllint)                                       |    3|     |                                                                          0|0.0s        |
|✅ YAML           |[prettier](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_prettier)                                    |   80|    2|                                                                          0|2.61s       |
|✅ YAML           |[v8r](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_v8r)                                              |   22|     |                                                                          0|45.3s       |
|✅ YAML           |[yamllint](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_yamllint)                                    |   81|     |                                                                          0|1.29s       |

See errors details in [**artifact MegaLinter reports** on CI Job page](https://github.com/oxsecurity/megalinter/actions/runs/3357733083)

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

**Comment by nvuillam** [comment]
## [🦙 MegaLinter](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files) status: :warning: [**WARNING**](https://github.com/oxsecurity/megalinter/actions/runs/3357746663)

|     Descriptor     |                                                                  Linter                                                                   |Files|Fixed|                                  Errors                                   |Elapsed time|
|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------|-----|----:|---------------------------------------------------------------------------|------------|
|✅ BASH             |[bash-exec](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/bash_bash_exec)                                  |    7|     |                                                                          0|0.02s       |
|✅ BASH             |[shellcheck](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/bash_shellcheck)                                |    7|     |                                                                          0|0.33s       |
|✅ BASH             |[shfmt](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/bash_shfmt)                                          |    7|    0|                                                                          0|0.06s       |
|✅ COPYPASTE        |[jscpd](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                         |2.98s       |
|✅ DOCKERFILE       |[hadolint](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/dockerfile_hadolint)                              |  105|     |                                                                          0|10.72s      |
|✅ JSON             |[eslint-plugin-jsonc](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/json_eslint_plugin_jsonc)              |   21|    0|                                                                          0|2.88s       |
|✅ JSON             |[jsonlint](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/json_jsonlint)                                    |   19|     |                                                                          0|0.3s        |
|✅ JSON             |[v8r](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/json_v8r)                                              |   21|     |                                                                          0|18.45s      |
|:warning: MARKDOWN  |[markdownlint](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/markdown_markdownlint)                        |  294|    2|[**5**](https://github.com/oxsecurity/megalinter/actions/runs/3357746663)  |6.51s       |
|✅ MARKDOWN         |[markdown-link-check](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/markdown_markdown_link_check)          |  294|     |                                                                          0|6.36s       |
|✅ MARKDOWN         |[markdown-table-formatter](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/markdown_markdown_table_formatter)|  294|    2|                                                                          0|8.49s       |
|✅ OPENAPI          |[spectral](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/openapi_spectral)                                 |    1|     |                                                                          0|1.09s       |
|:warning: PYTHON    |[bandit](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/python_bandit)                                      |  170|     |[**42**](https://github.com/oxsecurity/megalinter/actions/runs/3357746663) |3.5s        |
|✅ PYTHON           |[black](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/python_black)                                        |  170|    0|                                                                          0|5.21s       |
|✅ PYTHON           |[flake8](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/python_flake8)                                      |  170|     |                                                                          0|2.7s        |
|✅ PYTHON           |[isort](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/python_isort)                                        |  170|    0|                                                                          0|0.55s       |
|✅ PYTHON           |[mypy](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/python_mypy)                                          |  170|     |                                                                          0|10.72s      |
|✅ PYTHON           |[pylint](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/python_pylint)                                      |  170|     |                                                                          0|15.65s      |
|:warning: PYTHON    |[pyright](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/python_pyright)                                    |  170|     |[**272**](https://github.com/oxsecurity/megalinter/actions/runs/3357746663)|24.38s      |
|✅ REPOSITORY       |[checkov](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/repository_checkov)                                |yes  |     |no                                                                         |36.04s      |
|:warning: REPOSITORY|[devskim](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/repository_devskim)                                |yes  |     |[**59**](https://github.com/oxsecurity/megalinter/actions/runs/3357746663) |1.52s       |
|✅ REPOSITORY       |[dustilock](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/repository_dustilock)                            |yes  |     |no                                                                         |1.63s       |
|✅ REPOSITORY       |[git_diff](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/repository_git_diff)                              |yes  |     |no                                                                         |0.06s       |
|✅ REPOSITORY       |[secretlint](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/repository_secretlint)                          |yes  |     |no                                                                         |6.18s       |
|✅ REPOSITORY       |[syft](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/repository_syft)                                      |yes  |     |no                                                                         |3.58s       |
|✅ REPOSITORY       |[trivy](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/repository_trivy)                                    |yes  |     |no                                                                         |23.66s      |
|✅ SPELL            |[cspell](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/spell_cspell)                                       |  707|     |                                                                          0|27.38s      |
|✅ SPELL            |[misspell](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/spell_misspell)                                   |  533|    2|                                                                          0|0.58s       |
|✅ XML              |[xmllint](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/xml_xmllint)                                       |    3|     |                                                                          0|0.0s        |
|✅ YAML             |[prettier](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/yaml_prettier)                                    |   80|    0|                                                                          0|3.79s       |
|✅ YAML             |[v8r](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/yaml_v8r)                                              |   22|     |                                                                          0|55.46s      |
|✅ YAML             |[yamllint](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/descriptors/yaml_yamllint)                                    |   81|     |                                                                          0|1.83s       |

See errors details in [**artifact MegaLinter reports** on CI Job page](https://github.com/oxsecurity/megalinter/actions/runs/3357746663)

You could have the same capabilities but better runtime performances if you use a MegaLinter flavor:
- [**oxsecurity/megalinter/flavors/python@v5**](https://oxsecurity.github.io/megalinter/test-nuillam-cspell_files/flavors/python/) (56 linters)

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: docs/descriptors/spell_cspell.md [hunk 1]
```diff
@@ -20,20 +22,21 @@ Copy it at the root of your repository, read it, remove real spelling errors (af
...
-| SPELL_CSPELL_DISABLE_ERRORS              | Run linter but consider errors as warnings                                                                                                                                                                          | `false`                                         |
-| SPELL_CSPELL_DISABLE_ERRORS_IF_LESS_THAN | Maximum number of errors allowed                                                                                                                                                                                    | `0`                                             |
+| Variable | Description | Default value |
+| ----------------- | -------------- | -------------- |
+| SPELL_CSPELL_ANALYZE_FILE_NAMES | If set to `true`, MegaLinter will also send file names to cspell for analysis. Disable by defining `SPELL_CSPELL_ANALYZE_FILE_NAMES` to `false`` | `true` |
```
Comment (Kurt-von-Laven) [inline docs/descriptors/spell_cspell.md:27]: @nvuillam, an extra back tick snuck in there (c.f., [the rendered doc](https://oxsecurity.github.io/megalinter/6.14.0/descriptors/spell_cspell/#configuration-in-megalinter)).

Comment (nvuillam) [reply docs/descriptors/spell_cspell.md:27]: Good catch, thanks :)

**Pull request merged**